### ISSUE TITLE
Data dependency and server side rendering

### ISSUE THREAD

**Issue by xethorn** [yahoo/fluxible#13]
(sorry for today's spam, I've been trying to get a full functional app using flexible for the past few days.)

Questions: 
- how do you deal with multiple requests with fluxible to perform a complete a full server side rendering?
- how do you deal with global context (for instance: you have some user information you need to pass down to different stores so they can send the, let's say, user session to the api.)

Let's say you have the following chat application:

```
└── components
    ├── form.js
    ├── index.js
    ├── last_messages.js
    ├── thread.js
    └── sidebar.js
```

- Index: is performing an action to get the user informations.
- Sidebar: **needs** to have the list of channels to be rendered.
- Last messages: **needs** to have the list of the last messages that have been sent in all channels combined 
- Form: **need** to have additional user information to display (for instance) their logo.
- Thread: **need** an that is provided from the sidebar (or the url) to be rendered.

The [chat example](https://github.com/yahoo/flux-examples/tree/master/chat) shows that the data is being fetched within [server.js](https://github.com/yahoo/flux-examples/blob/master/chat/server.js#L36), but what happens when this request is actually triggered by a component? 

And more importantly, what happens when you have within one component, two components that requires data from different sources?

Advices would be much appreciated :)

**Comment by gpbl** [comment]
I'm also for keeping discrete actions, but how is this possible when rendering server-side? There, I rely on the route's `action` property to pre-fill the stores before rendering the page (see also #18). As solution, I'd go for skipping the route's `action` when on the client: 

```js
var detailRoute = {
  path: '/',
  method: 'get',
  page: 'detail',
  action: function (context, payload, done) {
    if (BROWSER) { 
       // let the single components start the actions
      done(); 
      return; 
    }
    // otherwise hydrate stores
    var detail = new Promise(function(resolve){
        context.executeAction(detailAction, {}, resolve);
    });
    var otherThing = new Promise(function(resolve){
        context.executeAction(thingAction, {}, resolve);
    });
    Promise.all([detail, otherThing]).then(function() { done() });
  }
```

As you see a route may call different actions to hydrate different stores. The thing is getting too complex and it doesn't feel right. For example, how do I know for each route which stores should be hydrated?

Now I'm creating a Promise and waiting for Promise.all(). Is there a way to wait for multiple actions? Could `executeAction` return a promise?

**Comment by gpbl** [comment]
Update: after some time working with fluxible, I found that skipping the route's action on the browser it is not a great idea. If I want to give the user an immediate feedback when changing a route, I can work with a RouteStore that listens to the `CHANGE_ROUTE_START` action.

Also, instead of using Promises and such, I've found [fluxible-action-utils](https://github.com/yahoo/fluxible-action-utils) helps to execute multiple actions and has also a nice way to catch callback errors.
